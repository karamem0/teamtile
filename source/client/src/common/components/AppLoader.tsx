//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { app } from '@microsoft/teams-js';

import { getAccessToken } from '../../managers/TokenManager';

import Presenter from './AppLoader.presenter';

interface AppLoaderProps {
  children?: React.ReactNode
}

function AppLoader(props: Readonly<AppLoaderProps>) {

  const { children } = props;

  const [ loading, setLoading ] = React.useState<boolean>(true);

  React.useEffect(() => {
    (async () => {
      try {
        setLoading(true);
        await app.initialize();
        await getAccessToken();
        app.notifyAppLoaded();
        app.notifySuccess();
      } catch (error) {
        app.notifyFailure({
          reason: app.FailedReason.AuthFailed,
          message: error instanceof Error ? error.message : Object.prototype.toString.call(error)
        });
      } finally {
        setLoading(false);
      }
    })();
  }, []);

  return (
    <Presenter loading={loading}>
      {children}
    </Presenter>
  );

}

export default AppLoader;
