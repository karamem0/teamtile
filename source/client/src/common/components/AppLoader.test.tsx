//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import AppLoader from './AppLoader.presenter';

test('create shapshot of when loading is true', async () => {
  const params = {
    children: <div data-testid="Children" />,
    loading: true
  };
  render(<AppLoader {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});

test('create shapshot of when loading is false', async () => {
  const params = {
    children: <div data-testid="Children" />,
    loading: false
  };
  render(<AppLoader {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});
