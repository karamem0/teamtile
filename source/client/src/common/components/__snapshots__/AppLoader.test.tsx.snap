// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot of when loading is false 1`] = `
<body>
  <div>
    <div
      data-testid="Children"
    />
  </div>
</body>
`;

exports[`create shapshot of when loading is true 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: gray;
}

@media (max-width: 599px) {
  .emotion-0 {
    min-height: calc(100vh - 0.5rem);
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    min-height: calc(100vh - 2rem);
  }
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        class="ui-loader"
        data-aa-class="Loader"
        data-uses-unhanded-props="true"
        role="progressbar"
      >
        <div
          class="ui-box ui-loader__indicator"
          data-uses-unhanded-props="true"
        >
          <div
            class="ui-box ui-loader__svg"
            data-uses-unhanded-props="true"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
